File: VB/Controllers/HomeController.vb
Imports System.Web.Mvc

Namespace EditorsModelBinding.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function Editors() As ActionResult
			Return View(ListsDataHelper.GetSampleCustomer())
		End Function

		<HttpPost>
		Public Function Editors(ByVal model As CustomerInfo) As ActionResult
			Return View("Details", model)
		End Function

		Public Function FormLayout() As ActionResult
			Return View(ListsDataHelper.GetSampleCustomer())
		End Function

		<HttpPost>
		Public Function FormLayout(ByVal model As CustomerInfo) As ActionResult
			Return View("Details", model)
		End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System.Web.Mvc;

namespace EditorsModelBinding.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult Editors() {
            return View(ListsDataHelper.GetSampleCustomer());
        }

        [HttpPost]
        public ActionResult Editors(CustomerInfo model) {
            return View("Details", model);
        }

        public ActionResult FormLayout() {
            return View(ListsDataHelper.GetSampleCustomer());
        }

        [HttpPost]
        public ActionResult FormLayout(CustomerInfo model) {
            return View("Details", model);
        }
    }
}